node/src/chain/acceptor.rs

```diff
@@ -171,25 +196,108 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> Acceptor<N, DB, VM> {
         Ok(())
     }
 
-    fn needs_update(blk: &Block, txs: &[SpentTransaction]) -> bool {
-        // Update provisioners at every epoch (where new stakes take effect)
-        if blk.header().height % EPOCH == 0 {
-            return true;
+    fn selective_update(
+        blk: &Block,
+        txs: &[SpentTransaction],
+        vm: &tokio::sync::RwLockWriteGuard<'_, VM>,
+        provisioners_list: &mut tokio::sync::RwLockWriteGuard<
+            '_,
+            ContextProvisioners,
+        >,
+    ) -> Result<()> {
+        let src = "selective";
+        let changed_prov = Self::changed_provisioners(blk, txs)?;
+        if changed_prov.is_empty() {
+            provisioners_list.remove_previous();
+        } else {
+            let mut new_prov = provisioners_list.current().clone();
+            for pk in changed_prov {
+                let is_stake = pk.is_stake();
+                let pk = pk.into_public_key();
+                let prov = pk.to_bs58();
+                info!(event = "get_provisioner", src, prov);
+                match vm.get_provisioner(pk.inner())? {
+                    Some(stake) if stake.value() >= MINIMUM_STAKE => {
+                        debug!(event = "new_stake", src, prov, ?stake);
```

Comment:
Do you mind using `info!` level for both `new_stake` and `remove_stake` log events instead? It should be useful to monitor this closely in ITN.

Reply:
Going to change it